polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -112,9 +112,31 @@ pub(super) fn ends_with(s: &Series, sub: &str) -> PolarsResult<Series> {
     let ca = s.utf8()?;
     Ok(ca.ends_with(sub).into_series())
 }
-pub(super) fn starts_with(s: &Series, sub: &str) -> PolarsResult<Series> {
-    let ca = s.utf8()?;
-    Ok(ca.starts_with(sub).into_series())
+pub(super) fn starts_with(s: &[Series]) -> PolarsResult<Series> {
+    let ca = &s[0].utf8()?;
+    let sub = &s[1].utf8()?;
+
+    match sub.len() {
+        1 => {
+            let out: BooleanChunked = match sub.get(0) {
+                Some(s) => ca.starts_with(s),
+                None => BooleanChunked::new(ca.name(), vec![None; ca.len()]),
```

Comment:
I think we should return `false` in this case. 

We should use `BooleanChunked::full(ca.name(), false, ca.len())` for that. This is much cheaper than going through vec as that is a double copy.